keel-web/src/main/kotlin/com/netflix/spinnaker/keel/services/ApplicationService.kt

```diff
@@ -154,14 +154,25 @@ class ApplicationService(
   ): ArtifactVersionSummary =
     when (artifact.type) {
       deb -> {
-        val appversion = AppVersion.parseName(version)
-        ArtifactVersionSummary(
+        var summary = ArtifactVersionSummary(
           version = version,
           environments = environments,
-          displayName = appversion?.version ?: version.removePrefix("${artifact.name}-"),
-          build = if (appversion != null) BuildMetadata(id = appversion.buildNumber.toInt()) else null,
-          git = if (appversion != null) GitMetadata(commit = appversion.commit) else null
+          displayName = version.removePrefix("${artifact.name}-")
         )
+
+        // attempt to parse helpful info from the appversion.
+        // todo: replace, this is brittle
+        val appversion = AppVersion.parseName(version)
+        if (appversion?.version != null) {
+          summary = summary.copy(displayName = appversion.version)
```

Comment:
Do we ever expect  fields of `appversion` to be null during normal operation? IF not, I'd put a `WARN` level log statement with the  value of`version` that's parsed to make it easier to debug in the future.

Reply:
we might fail, we might not fail if that's the case. it definitely puts us into a weird state. once we pull the appversion off the ami though it shouldn't be null. we don't rely on any of the other fields that i'm using here (if i remember correctly). Logging would probably be helpful, you're right.